include/libtorrent/disk_io_job.hpp

```diff
@@ -151,7 +154,12 @@ namespace libtorrent {
 		{
 			un() {}
 			// result for hash jobs
-			sha1_hash piece_hash;
+			struct hash_args
```

Comment:
did you mean for this to be a union? or is the `piece_hash` used at the same time as `block_hashes`?

Reply:
as far as I can tell, they're not used at the same time, but I just did a very shallow search.

Reply:
Both `piece_hash` and `block_hashes` are used at the same time in hash jobs which compute both v1 and v2 hashes in the same job.